File: 04web_iot/05cam.js
var express = require('express');
var cv = require('opencv');
var app = express();


var camera = new cv.VideoCapture(0);
//camera.setWidth(320);
//camera.setHeight(240);

app.use(express.static(__dirname + '/static'));
app.get('/', function(req, res){
	res.redirect('/cam.html');
});

app.get('/cctv', function(req, res){
	console.log(req.url, req.query);
    camera.read(function(err, im) {
        if (err){
        	console.log("Error", err);
        	res.sendStatus(500);
        }else{
        	console.log(im.size());
        	if (im.size()[0] > 0 && im.size()[1] > 0){
	        	var buff = im.toBuffer();
	        	var b64 = buff.toString('base64');
	        	console.log(buff.length);
	        	res.send(b64);
        	}else{
        		res.sendStatus(500);
        	}
        }
      });	
});

app.listen(5000, function(){
	console.log('server running on 5000.');
});

File: 04web_iot/05cam.py
from flask import Flask, redirect, request
import cv2
import base64

app = Flask(__name__)
app.config['SECRET_KEY'] = 'secret'

cam = cv2.VideoCapture(0)
cam.set(cv2.cv.CV_CAP_PROP_FRAME_WIDTH, 320)
cam.set(cv2.cv.CV_CAP_PROP_FRAME_HEIGHT, 240)

@app.route('/')
def main():
    return redirect('/static/cam.html')

@app.route('/cctv')
def oncctv():
    print 'cctv'
    ret, data = cam.read()
    if ret:
        ret, png = cv2.imencode('.png', data)
        b64 = base64.encodestring(png)
        return b64
    return 'camera is not ready.'
    
    
if __name__ == '__main__':
    app.run(host="0.0.0.0")
